core/modules/layout/layout.module

```diff
@@ -1543,13 +1542,22 @@ function layout_get_layout_info($layout_name = NULL) {
 /**
  * Load all layout context information from modules.
  *
+ * @param string $plugin_name
+ *   The name of the plugin. If no plugin is specified, all context info will
+ *   be returned.
+ *
  * @return array
  */
-function layout_get_context_info($plugin_name) {
+function layout_get_context_info($plugin_name = NULL) {
   $info = _layout_get_all_info('layout_context');
-  return isset($info[$plugin_name]) ? $info[$plugin_name] : FALSE;
-}
 
+  if (!$plugin_name) {
+    return $info;
+  }
+  else {
+    return isset($info[$plugin_name]) ? $info[$plugin_name] : FALSE;
```

Comment:
We understand that this function exists, `_layout_get_all_info('layout_context')`, however this function's name indicates that the function is private. We did not feel it was a good idea to have the views module call a private function buried in the layout module.

We chose to modify the function `layout_get_context_info($plugin_name = NULL) ` because we felt it was more in-line with what the function is doing, in this case retrieving context info. 

We also concluded that because the function `layout_get_layout_info($layout_name = NULL)` has $layout_name optional, it would be okay for `layout_get_context_info($plugin_name)` to have its argument optional as well.

No existing implementation of the function would break.

All that being said, if calling the private function aligns with Backdrop CMS practices, we'll happily make the adjustment. :)

Reply:
Layout is a required module so I think realistically "private" functions wouldn't really mean much. And yes, private functions do seem to get called from other files so IMO I think this is OK. Would welcome other opinions though.

Reply:
Thank you. This has been reverted back.